dm/pkg/binlog/event/util.go

```diff
@@ -408,6 +408,27 @@ func GetCharsetCodecByStatusVars(statusVars []byte) (encoding.Encoding, error) {
 	}
 }
 
+// GetTimezoneByStatusVars returns the timezone of upstream for "datetime" type alter table commands
+// For "ALTER TABLE ADD x datetime", not for "ALTER TABLE ADD x timestamp".
+func GetTimezoneByStatusVars(statusVars []byte, upstreamTZStr string) (string, error) {
+	vars, err := statusVarsToKV(statusVars)
+	b, ok := vars[QTimeZoneCode]
+
+	if !ok {
+		if err == nil {
+			// only happen when this is a dummy event generated by DM
```

Comment:
this comment should be updated, as in the comment of function some DDL will also not write timezone to status vars